picoquic/sender.c

```diff
@@ -3809,6 +3820,12 @@ int picoquic_prepare_packet_ready(picoquic_cnx_t* cnx, picoquic_path_t* path_x,
                             bytes_next = picoquic_format_available_stream_frames(cnx, bytes_next, bytes_max, &more_data, &is_pure_ack, &stream_tried_and_failed, &ret);
                         }
 
+                        /* Send bdp frames if there are no stream frames to send 
+                         * and if peer wishes to receive bdp frames */
+                        if(!cnx->client_mode && cnx->quic->default_bdp_option == 2 && (cnx->remote_parameters.enable_bdp == 2 || cnx->remote_parameters.enable_bdp == 3)) {
+                           bytes_next = picoquic_format_bdp_frame(cnx, bytes_next, bytes_max, path_x, &more_data, &is_pure_ack);
+                        }
+
```

Comment:
There is a better way to do that. Inside `picoquic_seed_ticket`, when new values of the CWIN are learned, the code places them in the server ticket hash table. At the same time, it could post a `miscellaneous frame` to queue transmission of the BDP frame.